client/Models/EditorModel.ts

```diff
@@ -0,0 +1,17 @@
+import { atom } from 'jotai';
+
+// Default Editor Configurations
+export const editorSettingsAtom = atom({
+    theme: 'monokai',
+    options: [
+        'enable-auto-indent',
+        'show-line-numbers',
+        'enable-live-autocompletion',
+        'enable-live-autocompletion',
+    ],
+    fontSize: 12,
+    softTabs: 4,
+    keybind: 'ace',
+    cursorStyle: 'ace',
+    foldStyle: 'manual',
+});
```

Comment:
This is not a model... this is actual data.  Add `lib/competitor-state.ts` with this.  That file will contain all state related to the competitor.